### PR TITLE
[#162] Checkbox Component

### PR DESCRIPTION
**Pull request by CMChemist**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
This PR adds a skeleton of the custom Checkbox component that the UI team designed. 

The Checkbox Component current accepts 4 properties:
- checked: boolean // (required) True=checked. False=unchecked
- setChecked: Function // (required) Expects a function that toggles the checked value
- size: number // (optional) Sets size of the checkbox. Default is 18
- disabled: booleon / (optional) True=disabled. False=enabled. Default is false.

I used CheckBox from React Native Elements to build this component, so I installed 'react-native-elements'.

**What is the current behavior? (You can also link to an open issue here)**
N/A

**What is the new behavior? (if this is a feature change)**
There is no new behavior at this time as the component is not currently being used anywhere in the app

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
No breaking changes.

Other Information:
The CheckBox component from 'react-native-elements' has a lot of useful properties that maybe help as we build out this component. Here's a link to the [React Native Elements CheckBox Docs](https://react-native-elements.github.io/react-native-elements/docs/checkbox.html) if you wanna read more about it.

### CODE REVIEW

File: src/elements/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { TouchableOpacity, TouchableWithoutFeedback } from 'react-native';
+import { CheckBox } from 'react-native-elements';
```
Comment (jacksonrya) [inline src/elements/Checkbox/Checkbox.tsx:3]: Unforunately, I think using this library and this imported component isn't a great choice for our checkbox. 14 / 20 props are irrelevant and/or redundant to what our checkbox needs to be and to how our checkbox can be implemented. If this was just a placeholder, `react-native` already comes with a CheckBox that could be used.

Though, it seems like a useful library for projects that are using other font and UI systems.

These two produce the same results:
```tsx
<TouchableHighlight
   onPress={() => { !disabled && handlePress(); }}
>
   <Icon
      name={isChecked ? 'arrowDown' : 'arrowUp'}
      color={color}
      size={size}
   />
</TouchableHighlight>

<CheckBox
   checkedIcon={<Icon name="arrowDown" color={color} size={size} />}
   uncheckedIcon={<Icon name="arrowUp" color={color} size={size} />}
   Component={disabled ? TouchableWithoutFeedback : TouchableHighlight}
   checked={isChecked}
   onPress={handlePress}
   size={size}
/>
```

**Comment by jacksonrya** [comment]
as for all the extra commits,

most likely, the branch that you branched off of in order to create `create-checkbox-component` was out of date. i would guess that you branched off of your local master? if you run `git status` on your local master, you'll probably see that you're behind in commits.

unless you have changes that you know you need to save, `run git fetch upstream && git reset --hard upstream/master` on your local master. this will sync it to our repo's current state (edited) 

in order to get those extra commits removed from your PR, you'll have to rebase onto upstream/master. during the rebase process, you can probably skip all of your commits before `ef983af`. you might even be able to just drop all of them if you're doing interactive rebasing.

notice that you have a handful of commits titled, "Merge branch 'master' of https://...banana-rn"
these are evidence that you ran `git pull` on your master branch, but that because your local master was out of sync, git had to merge the remote rather than fast-forward, which is what the merge mechanism tries to do first

File: src/elements/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (CMChemist) [reply src/elements/Checkbox/Checkbox.tsx:3]: I did see that react-native had a CheckBox component but the documentation online said that it was no longer supported, so I didn't think that made it a viable option. Should I change it to the react-native component anyways?

File: package.json [hunk 2]
```diff
@@ -26,10 +26,11 @@
     "expo-permissions": "~8.0.0",
     "metro-config": "^0.59.0",
     "react": "16.9.0",
-    "react-dom": "16.9.0",
+    "react-dom": "^16.9.0",
```
[the remark this answers is not in the document]
Comment (CMChemist) [reply package.json:29]: Weird, I didn't manually make that change. Must have happened when I ran npm install after installing `react-native-elements`.

File: package.json [hunk 3]
```diff
@@ -45,6 +46,7 @@
     "react-navigation-drawer": "2.3.3",
     "react-navigation-hooks": "1.1.0",
     "react-navigation-stack": "1.10.3",
+    "styled-components": "^5.1.0",
```
[the remark this answers is not in the document]
Comment (CMChemist) [reply package.json:49]: Oh sorry, that was install when I was trying a different way to implement the checkbox. I'll remove it.

File: src/elements/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+interface CheckboxProps {
+	checked: boolean;
+	setCheck: Function;
```
[the remark this answers is not in the document]
Comment (CMChemist) [reply src/elements/Checkbox/Checkbox.tsx]: Fixed in most recent commit.

File: src/util/colors.ts [hunk 5]
```diff
@@ -10,4 +10,4 @@ export const LIGHT_GRAY = '#F0EEEE';
 export const LIGHT_GRAY_DISABLED = '#C4C4C4';
 export const DARK_GRAY = 'rgb(60, 60, 60)';
 export const DARK_GRAY_TRANSPARENT = 'rgba(60, 60, 60, .5)';
-export const RED = '#E10000';
+export const RED = '#E10000';
```
[the remark this answers is not in the document]
Comment (CMChemist) [reply src/util/colors.ts]: Reverted to upstream/master commit.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (CMChemist) [reply package.json]: Uninstalled "styled-components"

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (CMChemist) [reply package.json]: Corrected.

File: src/elements/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (jacksonrya) [reply src/elements/Checkbox/Checkbox.tsx:3]: Oh nice find on the deprecation, I didn't see that in the in-file documentation. That would have just been a convenient placeholder, all without needing to add an extra dependency.

I think that a component structure that just uses the native `TouchableHighlight` and our custom Icon component is a better choice. I don't see the need to depend upon an entire library if we can accomplish it on our own and with only a handful of lines of code.

File: package.json [hunk 6]
```diff
@@ -62,7 +63,7 @@
     "eslint-plugin-react": "^7.17.0",
     "eslint-plugin-react-hooks": "^1.7.0",
     "react-native-svg-transformer": "^0.14.3",
-    "typescript": "^3.6.3"
+    "typescript": "^3.8.3"
```
Comment (jacksonrya) [inline package.json:66]: was there a reason to bump the typescript version?

Comment (CMChemist) [reply package.json:66]: Oh when I was I was watching the CI / build (pull_request) checks run I got a notification that my typescript was out of date so I ran npm i typescript to update it. Should I revert it back to "^3.6.3"?

Comment (jacksonrya) [reply package.json:66]: ya i think you should revert. though it's doubtful updating typescript will break anything, it's good practice to separate these concerns.

we can make note of it though for a future PR.

Comment (CMChemist) [reply package.json]: Reverted the file to before the typescript update.

**Comment by jacksonrya** [comment]
Make sure to re-run `npm install` after you change the package.json to make sure the package-lock file is in sync.

Additionally, despite you force-pushing after, what I assume, was a rebase, the commits remain on the Github PR which means you didn't actually remove them from your commit history.

**Pull request merged**